egi_notebooks_hub/d4science.py

```diff
@@ -151,3 +152,193 @@ def pre_spawn_start(self, user, spawner):
 
     def get_handlers(self, app):
         return [(r"/login", self.login_handler)]
+
+
+class D4ScienceContextHandler(OAuthLoginHandler):
+    def get_state(self):
+        context = self.get_argument("context", None)
```

Comment:
in case there i no context passed we should report an error i guess

Reply:
It's raised in https://github.com/EGI-Federation/egi-notebooks-hub/pull/53/files/66d6db649872adc623b2047a82026f7c70504d13#diff-92d763427bc3f7bb3f8f82693509678ae73c4447e74cf44034459b6a5e164d7cR205